client/components/chip/index.tsx

```diff
@@ -1,14 +1,25 @@
 /** @format **/
+import React from 'react';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 import { HoverTooltip } from 'wcpay/components/tooltip';
 
-const types = [ 'primary', 'light', 'warning', 'alert' ];
+const types = [ 'primary', 'light', 'warning', 'alert' ] as const;
 
-const Chip = ( props ) => {
+export type ChipType = typeof types[ number ];
```

Comment:
I think we can simplify this a bit, making `ChipType` directly a template literal, and `type` optional. Then we don't need to use `find` later, just the provided value. Something like:

```tsx
export type ChipType = 'primary' | 'light' | 'warning' | 'alert';

interface Props {
    // ...
	type?: ChipType;
    // ...
}

// ...
	const classNames = [
		'chip',
		`chip-${ type || 'primary' }`,
// ...
```

The only downside I see, is not using `primary` style when an invalid type is provided from JS code.

Reply:
The problem with changing

```ts
`chip-${ types.find( ( t ) => t === type ) || 'primary' }`,
```

to 

```ts
`chip-${ type ?? 'primary' }`,
```

Is that if someone now passes in an invalid string as the type (e.g. `invalid` which is one of the unit tests) you'll get a class that's `chip-<invalid_type>` instead of the expected `chip-primary`.

To address that we can either keep the `const types = [ /* ... */ ];` array and the `types.find( /* ... */ )` functionality, or add a type guard like

```ts
const isChipType = ( type?: string ): type is ChipType => {
	return types.includes( type as ChipType );
};

// ...

const classNames = [
  // ...
  `chip-${ isChipType( type ) ? type : 'primary' }`,
  // ...
];
```

The type guard may be more TypeScript-y, but I'm fine with keeping the `.find` approach as well 🤔 

What do you think? I've pushed be9a333ac9240d245a0559db8049c5d5dba83c35 as an example of the "more TS-y" approach.

Reply:
I like type guards, but you're right it could be too "TS-y" 😄 Maybe that's too much if we want to keep the TS migration barrier low enough.